src/state/store.ts

```diff
@@ -0,0 +1,28 @@
+import { Action, combineReducers, configureStore, Dispatch } from '@reduxjs/toolkit';
+import { legendReducer } from './legendSlice';
+
+const rootReducer = combineReducers({
+  // tooltip: tooltipReducer,
+  legend: legendReducer,
+});
+
+export const createStore = (chartName: string = 'Chart') => {
+  console.log('createStore', chartName);
+  return configureStore({
+    reducer: rootReducer,
+    middleware: getDefaultMiddleware =>
+      getDefaultMiddleware({
+        serializableCheck: {
+          ignoredActions: ['legend/addSupplier', 'legend/removeSupplier'],
+          ignoredPaths: ['legend.payloadSuppliers'],
+        },
+      }),
+    devTools: {
+      serialize: true,
+      name: 'recharts-' + chartName,
```

Comment:
This is useful for debugging. It reuses already present `id` and `chartName` props.